scripts/scr_squads/scr_squads.gml

```diff
@@ -8,14 +8,22 @@ the requested squad type , if the squad is not possible it will  not be made*/
 
 function create_squad(squad_type, company, squad_loadout = true){
 	var squad_unit_types, fulfilled,unit, squad, squad_unit;
-
+	show_debug_message("1")
 	var squad_count = array_length(obj_ini.squads);
 	var fill_squad =  obj_ini.squad_types[$ squad_type];			//grab all the squad struct info from the squad_types struct
 	var squad_fulfilment = {};		
 	squad_unit_types = struct_get_names(fill_squad);		//find out what type of units squad consists of
-	for (var i = 0;i < array_length(squad_unit_types);i++){
+	var unit_type_count = array_length(squad_unit_types);		
+	for (var i = 0;i < unit_type_count;i++){
+		if (squad_unit_types[i] == "display_name"){
```

Comment:
this if is responsible for trimming the `squad_unit_types` array?

Reply:
yeah may fiddle this to a slightly snappier fix at some point